### ISSUE TITLE
Bug or feature? assigning empty array to csr matrix row doesn't work.

### ISSUE THREAD

**Issue by xuanqing94** [scipy/scipy#13126]
Let's say I want to set one row of a csr matrix A to all zeros. Of course I can do A[0, :] = 0,
but sometimes I do A[0, :] = a where a is an empty csr row. Surprisingly, the resulting A doesn't change!!

But if I do a small change to a, such as a[0] = 0.00001, and set A[0, :] = a, then the 0-th row of A will be (close) to zeros.

I am omitting the code below because I believe the developers know about this phenomenon. Just posted here because I am confused why this happened.

I am using Scipy 1.4.1 + numpy 1.17.0
 
#### Reproducing code example:

```
Sample code to reproduce the problem
```

#### Error message:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  ...
```

**Comment by m-richards** [comment]
I belive you are describing the same issue as in #12819, which was fixed in Scipy 1.5.3.

**Comment by xuanqing94** [comment]
@m-richards Thank you! after upgrading to 1.5.4 it works well.
